feat: enhance OpenAI and Azure OpenAI authentication support

KrtinShet opened the pull request:
- Updated documentation in README.md and authentication.md to include Azure OpenAI configuration options.
- Modified CLI to accept Azure OpenAI parameters and handle them appropriately.
- Enhanced OpenAIKeyPrompt component to support Azure OpenAI input fields.
- Updated OpenAIContentGenerator to differentiate between Standard OpenAI and Azure OpenAI configurations.
- Added environment variable checks for Azure OpenAI in the sandbox and configuration files.

## TLDR

<!-- Add a brief description of what this pull request changes and why and any important things for reviewers to look at -->

## Dive Deeper

<!-- more thoughts and in-depth discussion here -->

## Reviewer Test Plan

<!-- when a person reviews your code they should ideally be pulling and running that code. How would they validate your change works and if relevant what are some good classes of example prompts and ways they can exercise your changes -->

## Testing Matrix

<!-- Before submitting please validate your changes on as many of these options as possible -->

|          | 🍏  | 🪟  | 🐧  |
| -------- | --- | --- | --- |
| npm run  | ❓  | ❓  | ❓  |
| npx      | ❓  | ❓  | ❓  |
| Docker   | ❓  | ❓  | ❓  |
| Podman   | ❓  | -   | -   |
| Seatbelt | ❓  | -   | -   |

## Linked issues / bugs

<!--
Link to any related issues or bugs.

**If this PR fully resolves the issue, use one of the following keywords to automatically close the issue when this PR is merged:**

- Closes #<issue_number>
- Fixes #<issue_number>
- Resolves #<issue_number>

*Example: `Resolves #123`*

**If this PR is only related to an issue or is a partial fix, simply reference the issue number without a keyword:**

*Example: `This PR makes progress on #456` or `Related to #789`*
-->

KrtinShet committed:
feat: enhance OpenAI and Azure OpenAI authentication support

- Updated documentation in README.md and authentication.md to include Azure OpenAI configuration options.
- Modified CLI to accept Azure OpenAI parameters and handle them appropriately.
- Enhanced OpenAIKeyPrompt component to support Azure OpenAI input fields.
- Updated OpenAIContentGenerator to differentiate between Standard OpenAI and Azure OpenAI configurations.
- Added environment variable checks for Azure OpenAI in the sandbox and configuration files.

Mingholy commented:
Thanks for the contribution!

IMHO, several optimizations can be made to the current implementation:
1.  Add Azure OpenAI as a new `AuthType`, along with a new option in AuthDialog;
2. Implement a new `ContentGenerator` that wraps all the Azure-specific code;
3. According to [Create and authenticate a AzureOpenAI](https://learn.microsoft.com/en-us/javascript/api/overview/azure/openai-readme?view=azure-node-latest#create-and-authenticate-a-azureopenai), it is recommended to authenticate clients using a token provider.

krtinshet8 committed:
fix: address PR feedback for Azure OpenAI support

- Implemented Azure OpenAI authentication in validateNonInterActiveAuth.ts and auth.ts.
- Updated AuthDialog and OpenAIKeyPrompt components to handle Azure OpenAI input fields and prompts.
- Introduced AzureOpenAIContentGenerator for Azure-specific content generation.
- Enhanced content generator configuration to support Azure OpenAI parameters.
- Updated logging to exclude Azure OpenAI responses and errors for better security.

KrtinShet commented:
I've made the requested changes. Please review again.

Mingholy reviewed (changes requested):
Minor changes are needed, but overall, it is looking good.

krtinshet8 committed:
fix(cli): default OpenAIKeyPrompt to 'openai' when variant is omitted to match tests and expected UX

KrtinShet commented:
Please review again.

Mingholy reviewed (changes requested):
I did some testing and found some additional issues. I think these will have some impact on the user experience, and you can verify this as well. Please fix them when you have time.

krtinshet8 committed:
fix: address Azure OpenAI PR review comments and improve implementation

krtinshet8 committed:
Merge main into feat/azure-openai-support

wenshao commented on packages/cli/src/utils/sandbox.ts:
```diff
@@ -820,8 +820,308 @@ export async function start_sandbox(
             `Sandbox process exited with code: ${code}, signal: ${signal}`,
           );
         }
+<<<<<<< HEAD
```
**[Critical]** This file still contains unresolved Git merge-conflict markers (`<<<<<<<`, `=======`, `>>>>>>>`).

That leaves the CLI package unparsable and breaks typecheck/build before the Azure changes can be exercised. Please resolve the merge and remove the conflict block entirely.

_— gpt-5.4 via Qwen Code /review_

wenshao commented on packages/core/src/core/geminiChat.ts:
```diff
@@ -188,6 +188,91 @@ export class GeminiChat {
     validateHistory(history);
   }
 
+<<<<<<< HEAD
```
**[Critical]** This file still contains unresolved Git merge-conflict markers (`<<<<<<<`, `=======`, `>>>>>>>`).

That makes the core package fail parsing/typecheck immediately. Please finish the merge here and keep only the intended final implementation.

_— gpt-5.4 via Qwen Code /review_

wenshao commented on packages/core/src/core/azureOpenAIContentGenerator.ts:
```diff
@@ -0,0 +1,59 @@
+ */
+
+import { Config } from '../config/config.js';
+import { OpenAIContentGenerator } from './openaiContentGenerator.js';
```
**[Critical]** `./openaiContentGenerator.js` does not exist in this package layout. The existing OpenAI implementation lives under `./openaiContentGenerator/…`, so this import leaves `npm run build` failing with `Could not resolve "./openaiContentGenerator.js"`.

Please import from the real module path that already exists in `packages/core/src/core/openaiContentGenerator/`.

_— gpt-5.4 via Qwen Code /review_

wenshao commented on packages/cli/src/ui/components/OpenAIKeyPrompt.tsx:
```diff
@@ -9,67 +9,121 @@ import { useState } from 'react';
+          return;
+        } else if (currentField === 'azureApiKey') {
+          setCurrentField('azureApiVersion');
+          return;
```
**[Suggestion]** The interactive Azure flow only accepts `azureApiKey` and requires it before submit, even though the runtime/validation code in this PR also supports `AZURE_OPENAI_BEARER_TOKEN`.

That makes the interactive setup path inconsistent with the implemented auth capability. Either add bearer-token input/support here, or drop bearer-token support from the runtime/docs until the UI path exists.

_— gpt-5.4 via Qwen Code /review_
